test/spec/com/cinergix/mapper/ObjectMapperTest.java

```diff
@@ -0,0 +1,818 @@
+			
+			ResultSet result = dbHelper.getResultSetForQuery( "SELECT id as user_id, name as user_name, email as user_email, age as user_age, weight as user_weight, dob as user_dob, last_update as user_last_update FROM user WHERE id LIKE 'testID'" );
+			result.next();
+			
+			method.setAccessible(true);
+			Date dob = (Date)( method.invoke( mapper, result, "user_last_update", Date.class ) );
+			method.setAccessible(false);
+			
+		    Calendar cal = Calendar.getInstance();
+		    cal.setTime( dob );
+		    int year = cal.get(Calendar.YEAR);
+		    int month = cal.get(Calendar.MONTH);
+		    int day = cal.get(Calendar.DAY_OF_MONTH);
+		    int hour = cal.get(Calendar.HOUR_OF_DAY);
+		    int min = cal.get(Calendar.MINUTE);
+		    int sec = cal.get(Calendar.SECOND);
+			
+			assertEquals( "parseValue should convert the given Timestamp Object to Date if type is Date class", 2015, year );// In Java Calendar month is starting from 0
+			assertEquals( "parseValue should convert the given Timestamp Object to Date if type is Date class", ( 10 - 1 ), month );
+			assertEquals( "parseValue should convert the given Timestamp Object to Date if type is Date class", 22, day );
+			assertEquals( "parseValue should convert the given Timestamp Object to Date if type is Date class", 8, hour );
+			assertEquals( "parseValue should convert the given Timestamp Object to Date if type is Date class", 30, min );
+			assertEquals( "parseValue should convert the given Timestamp Object to Date if type is Date class", 20, sec );
+			
+		}catch ( NoSuchMethodException e1 ){
+			e1.printStackTrace();
+		}catch ( IllegalAccessException e2 ){
+			e2.printStackTrace();
+		}catch( InvocationTargetException e3 ){
+			e3.printStackTrace();
+		}catch ( SQLException sqlex) {
+			 sqlex.printStackTrace();
+		}
+	}
+	
+	@Test( expected = DataTypeConversionException.class )
+	public void parseValueShouldThrowDataTypeConversionExceptionWhenTryingToConvertToIncompetableTypes(){
+		try{
+			
+			ObjectMapper<UserMock> mapper = new ObjectMapper<UserMock>();
+			
+			Class mapperClass = ObjectMapper.class;
+			Method method = mapperClass.getDeclaredMethod( "parseValue", ResultSet.class, String.class, Class.class );
+			
+			ResultSet result = dbHelper.getResultSetForQuery( "SELECT id as user_id, name as user_age, email as user_email FROM user WHERE id LIKE 'testID'" );
+			result.next();
+			
+			List<UserMock> list = mapper.mapResultSetToObject( result, UserMock.class );
+			
+		}catch ( NoSuchMethodException e1 ){
+			e1.printStackTrace();
+		}catch ( SQLException sqlex) {
+			 sqlex.printStackTrace();
+		}
+	}
+	
+	
+	/****************************************************************************************
+	 * Test getFieldColumnMapping method													*
+	 ****************************************************************************************/
+	@Test
+	public void getFieldColumnMappingShouldReturnMappedSetOfFieldVsColumnNameFromTheAnnotationInHashMap(){
+		try{
+			
```

Comment:
Create before/after hooks to do common reflection related stuff to setup the test.
Use a class level definition for the class that is being tested ```static final Class classBeingTested = ObjectMapper.class```